### PR TITLE
[Feat - ADF 736]: Incorporate in items the metadata properties restrictions

### PR DESCRIPTION
**Pull request by habemuscode**
**Ticket**

https://oat-sa.atlassian.net/browse/ADF-736

### CODE REVIEW

**Review by hectoras — APPROVED** [review]

File: views/cypress/tests/manage-schema.spec.js [hunk 1]
```diff
@@ -148,6 +148,10 @@
             cy.renameSelectedNode(selectors.itemForm, selectors.editItemUrl, childItemName);
         });
 
+        it('appears error on save due to notEmpty restriction', function () {
+            cy.get('div[class="form-error"]').should('have.text', 'This field is required');
+        });
```
Comment (hectoras) [inline views/cypress/tests/manage-schema.spec.js:153]: nitpick: I would suggest to reword the test name so it reads as a English sentence ("it shows an error on save due to notEmpty restriction"). BTW I'm used to that kind of phrasing on tests, so maybe it is just a matter of personal preference :)

```suggestion
        it('shows an error on save due to notEmpty restriction', function () {
            cy.get('div[class="form-error"]').should('have.text', 'This field is required');
        });
```

**Review by hectoras — APPROVED** [review]

**Review by 404passionFruit — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request's target is not `master`
- [x] Pull request title and description are meaningful
- [x] Commits are following conventional commits
- [x] Kitchen links are not exposed in the description

File: views/cypress/tests/manage-schema.spec.js [hunk 2]
```diff
@@ -90,6 +90,30 @@
                 selectors.editClassUrl
             );
         });
+
+        it('validate restriction - notEmpty', function () {
+            cy.getSettled('span[class="icon-edit"]').last().click();
```
Comment (clagosarias) [inline views/cypress/tests/manage-schema.spec.js:95]: Maybe we can group up this in a command?

File: views/cypress/tests/manage-schema.spec.js [hunk 3]
```diff
@@ -124,6 +148,10 @@
             cy.renameSelectedNode(selectors.itemForm, selectors.editItemUrl, childItemName);
         });
 
+        it('appears error on save due to notEmpty restriction', function () {
```
Comment (clagosarias) [inline views/cypress/tests/manage-schema.spec.js:151]: Two other restriction validations are pending (formFieldOrder and languageDependant)

**Review by clagosarias — CHANGES_REQUESTED** [review]

**Review by 404passionFruit — APPROVED** [review]

File: views/cypress/tests/manage-schema.spec.js (hunk 2 above)
Comment (habemuscode) [reply views/cypress/tests/manage-schema.spec.js]: Added

**Comment by clagosarias** [comment]
@habemuscode  Test is failing for me:

Screenshot 2021-11-26 at 10 16 04
Screenshot 2021-11-26 at 10 20 47

**Review by clagosarias — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [ ]  New code is not subject to concurrency issues (if applicable)
- [ ]  The feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request's target is not master
- [x]  Pull request title and description are meaningful
- [x]  Commits are following conventional commits
- [x]  Kitchen links are not exposed in the description

**Pull request merged**